be/src/exprs/string_functions.cpp

```diff
@@ -516,17 +516,17 @@ StringVal StringFunctions::concat_ws(
         return strs[0];
     }
 
-    if (strs[0].is_null) {
+    if (strs[0].is_null && 0 == sep.len) {
```

Comment:
NULL is different with empty string. 

For NULL, this function return NULL; for empty string, return NOT NULL

you can reference https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_concat-ws

Reply:
Fix done, Sorry~